File: problemas/iniciante/3147/3147.py
def main():
    H, E, A, O, W, X = map(int, input().split())
    ladoBem = H + E + A + X
    ladoMal = O + W

    print("Middle-earth is safe." if (ladoBem > ladoMal and ladoMal < ladoBem) else "Sauron has returned.")

main()

File: problemas/iniciante/3147/3147.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";

process.stdin.on("data", (chunk) => {
    input += chunk;
});

process.stdin.on("end", () => {
    let lines = input.split('\n');
    const [H, E, A, O, W, X] = lines.shift().trim().split(' ').map(Number);

    const ladoBem = H + E + A + X;
    const ladoMal = O + W;
    
    console.log((ladoBem > ladoMal && ladoMal < ladoBem) ? "Middle-earth is safe." : "Sauron has returned.");

});
